frappe/desk/form/save.py

```diff
@@ -59,6 +59,17 @@ def cancel(doctype=None, name=None, workflow_state_fieldname=None, workflow_stat
 	frappe.msgprint(frappe._("Cancelled"), indicator="red", alert=True)
 
 
+@frappe.whitelist()
+def discard(doctype: str, name: str | int):
+	"""discard a draft document"""
+	doc = frappe.get_doc(doctype, name)
+	capture_doc(doc, "Discard")
+
+	doc.discard()
```

Comment:
One concern is clash with existing fieldnames/method :skull: 

Nothing in our codebase, but can happen in wild.